README.md
![alt XOOPS CMS](https://xoops.org/images/logoXoops4GithubRepository.png)
## LatestNews module for  [XOOPS CMS 2.5.9+](https://xoops.org)
[![XOOPS CMS Module](https://img.shields.io/badge/XOOPS%20CMS-Module-blue.svg)](https://xoops.org)
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](http://www.gnu.org/licenses/gpl-2.0.html)

**Latest News Block** for [XOOPS CMS](https://xoops.org) is for News module

Current and upcoming "next generation" versions of XOOPS CMS are being crafted on GitHub at: https://github.com/XOOPS



docs/changelog.txt
<h5>0.74 Beta 1   [WORK IN PROGRESS - NOT RELEASED]</h5>  Dev: XOOPS 2.5.10, PHP 7.3.7
<hr>
- namespaces (mamba)
- class autoloading (mamba)
WORK IN PROGRESS


<h5>0.73 Beta 2019-08-12</h5>
<hr>
- moved all images, CSS, and JS files to /assets (mamba)
- renamed .html Smarty templates to .tpl (mamba)
- removed additional pass-by-references instances (zyspec)
- fixed _AM_ defines (added / deleted / corrected) (zyspec)
- in ./admin/blocksadmin.php removed unnecessary check for $xoopsUser->isAdmin() (zyspec)
- default language to 'english' if language file(s) not found in ./admin/blocksadmin.php (zyspec)
- removed functions in ./blocksadmin.php and put code 'in-line' (zyspec)
- very minor block template cleanup (zyspec)

- @todo: replace HTML marquee tag using CSS translate[X|Y] or keyframes


<h5>0.73 Beta 1</h5>
<hr>
updated for News 1.70 database structure (mamba)
converted to XOOPS 2.5.6 GUI (mamba)


<h5>0.72</h5>
<hr>
Added auto height feature by set 0 in height option
Replaced default value of featured news option to 0
Fixed marquee mouseover
Fixed WYSIWYG edited content bug


<h5>0.7</h5>
<hr>
Added clone option for XOOPS 2.0.x
Added permissions panel
Added hide/show option for print/pdf/email icons
Added hide/show option for More News link
Added link and description to each image automatically for SEO.
Added Scroll ticker option with height and speed control
added featured news option
fixed display all articles with no topic selected ( by bandit-x )
fixed css conflict when select news module as main module
Replace edit and delete links with icons


<h5>0.6</h5>
<hr>
Fixed bug: block not working if outside a news module page ( by bandit-x and trabis )
Added a check to prevent errors if the news module is not installed and active ( by trabis )
Added option multiselect topics ( by trabis )
Added option 'sort by' ( by trabis )
Added limit to "More News" link ( by young )
Added danish translation ( by anderssk )


<h5>0.5 ( First Public Release )</h5>
<hr>
Added option to truncate text without cutting the last word ( by trabis )
Added image position option
Refactored blocks language constants
Fixed css bug ( by taimur )
Fixed many bugs ( by trabis and bandit-x )


<h5>0.4</h5>
<hr>
Added block admin page ( by bandit-x )
Added admin links for edit and delete
Added read counter with it's show/hide option
Fixed date function ( by bandit-x )
Fixed comments counter


<h5>0.3</h5>
<hr>
Added number of columns option
Added hide/show option for topic title
Added autothumbnail with image width/height options
Added border and border color options


<h5>0.2</h5>
<hr>
Added columns to display news list ( by bandit-x )
Added hide/show options for posted by, author and date
Added hide/show option for comments


<h5>0.1</h5>
<hr>
Initial release based on last article block


xoops_version.php
<?php
// ######################################################################
// #                                                                    #
// # Latest News block by Mowaffak ( www.arabxoops.com )                #
// # based on Last Articles Block by Pete Glanz (www.glanz.ru)          #
// # Thanks to:                                                         #
// # Trabis ( www.xuups.com ) and Bandit-x ( www.bandit-x.net )         #
// #                                                                    #
// ######################################################################
// # Use of this program is goverened by the terms of the GNU General   #
// # Public License (GPL - version 1 or 2) as published by the          #
// # Free Software Foundation (http://www.gnu.org/)                     #
// ######################################################################
// defined('XOOPS_ROOT_PATH') || die('Restricted access');

require_once __DIR__ . '/preloads/autoloader.php';

$modversion['version']       = '0.74';
$modversion['module_status'] = 'Beta 1';
$modversion['release_date']  = '2019/08/13';
$modversion['name']          = _MI_LATESTNEWS_BLOCK;
$modversion['description']   = _MI_LATESTNEWS_BLOCKS_DESC;
$modversion['credits']       = 'Original version Darren Poulton<br>paladin@intaleather.com.au<br>http://paladin.intaleather.com.au/<br>Ported by: Sylvain B. (sylvain@123rando.com)<br>Altered version SUIN<br>http://tms.s10.xrea.com:8080/';
$modversion['author']        = 'Mowaffak, Trabis, Bandit-x, Zyspec, Mamba';
$modversion['help']          = 'page=help';
$modversion['license']       = 'GNU GPL 2.0 or later';
$modversion['license_url']   = 'www.gnu.org/licenses/gpl-2.0.html';
$modversion['official']      = 0; //1 indicates supported by XOOPS Dev Team, 0 means 3rd party supported
$modversion['image']         = 'assets/images/logoModule.png';
$modversion['dirname']       = basename(__DIR__);
$modversion['modicons16']    = 'assets/images/icons/16';
$modversion['modicons32']    = 'assets/images/icons/32';

//about
$modversion['module_website_url']  = 'www.xoops.org';
$modversion['module_website_name'] = 'XOOPS';
$modversion['min_php']             = '5.5';
$modversion['min_xoops']           = '2.5.10';
$modversion['min_admin']           = '1.2';
$modversion['min_db']              = ['mysql' => '5.5'];

$modversion['hasAdmin']    = 1;
$modversion['system_menu'] = 1;
$modversion['adminindex']  = 'admin/index.php';
//$modversion['adminindex'] = "admin/myblocksadmin.php";
$modversion['adminmenu'] = 'admin/menu.php';

// ------------------- Help files ------------------- //
$modversion['helpsection'] = [
    ['name' => _MI_LATESTNEWS_OVERVIEW, 'link' => 'page=help'],
    ['name' => _MI_LATESTNEWS_DISCLAIMER, 'link' => 'page=disclaimer'],
    ['name' => _MI_LATESTNEWS_LICENSE, 'link' => 'page=license'],
    ['name' => _MI_LATESTNEWS_SUPPORT, 'link' => 'page=support'],
];

// Blocks
$modversion['blocks'][1]['file']      = 'latestnews.php';
$modversion['blocks'][1]['name']      = _MI_LATESTNEWS_BLOCK;
$modversion['blocks'][1]['show_func'] = 'block_latestnews_show';
$modversion['blocks'][1]['edit_func'] = 'b_latestnews_edit';
$modversion['blocks'][1]['template']  = 'block_latestnews.tpl';
$modversion['blocks'][1]['options']   = '6|2|200|100|100|2|dcdcdc|0|right|1|1|1|1|1|1|1|1|1|1|1|1|0|100|30|published|';
$modversion['blocks'][1]['can_clone'] = true;

// On Update
if (!empty($_POST['fct']) && !empty($_POST['op']) && !empty($_POST['diranme']) && 'modulesadmin' === $_POST['fct']
    && 'update_ok' === $_POST['op']
    && $_POST['dirname'] == $modversion['dirname']) {
    require_once __DIR__ . '/include/onupdate.inc.php';
}
